packages/wallets/src/evm/wallets/smart-wallet.ts

```diff
@@ -76,6 +78,90 @@ export class SmartWallet extends AbstractClientWallet<
     return this.connector?.personalWallet;
   }
 
+  /**
+   * @returns the signature of the message
+   */
+  public async signMessage(message: Bytes | string): Promise<string> {
+    const erc4337Signer = await this.getSigner();
+    const connector = await this.getConnector();
+
+    // Deploy smart wallet if not already deployed.
+    const isDeployed = await connector.isDeployed();
+    if (!isDeployed) {
+      console.log(
+        "Account contract not deployed yet. Deploying account before signing message",
+      );
+      const tx = await erc4337Signer.sendTransaction({
+        to: await this.getAddress(),
+        data: "0x",
+      });
+      await tx.wait();
+    }
+
+    // Get underlying signer
+    let signer = erc4337Signer;
+    if ((signer as any).originalSigner) {
+      signer = (signer as any).originalSigner;
+    }
+
+    const chainId = await erc4337Signer.getChainId();
+    const address = await connector.getAddress();
+    const AccountMessage = [{ name: "message", type: "bytes" }];
+
+    /**
+     * We first try to sign the EIP-712 typed data i.e. the message mixed with the smart wallet's domain separator.
+     * If this fails, we fallback to the legacy signing method.
+     */
+    try {
+      const signature = await (
+        signer as providers.JsonRpcSigner
+      )._signTypedData(
```

Comment:
can we use `signTypedDataInternal()` instead ? already handles a lot of this for you and its better to centralize this logic in one place

Reply:
I tried importing it but that didn't work; I'll try again.